compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1969,10 +1969,19 @@ impl<'a: 'ast, 'ast, 'tcx> LateResolutionVisitor<'a, '_, 'ast, 'tcx> {
             && let ast::ExprKind::Path(None, _) = lhs.kind
         {
             if !ident_span.from_expansion() {
+                let (span, text) = match self.r.tcx.sess.source_map().span_to_snippet(ident_span) {
```

Comment:
It's not necessary to go to the source map here.
`ast::ExprKind::Path(None, path)` above already contains the identifier in `path`.
`ident_span` should also be there, no need to pass it.

Reply:
sorry, I don't quite understand. could you be more specific? I want a `String` for the lhs identifier to judge whether it start with `let_`, can we get that `String` from `path` in `ast::ExprKind::Path(None, path)`?  browse the codebase but can't find the expectant api

Reply:
`path.segments.first()` will give the first path segment, `segment.ident.as_str()` will give its string representation, `segment.ident.span` will give its span.